R/zchunk_LA144.building_det_flsp.R

```diff
@@ -46,60 +48,380 @@ module_energy_LA144.building_det_flsp_DISABLED <- function(command, ...) {
+      summarise(`1980` = sum(`1980`),
+                `1990` = sum(`1990`)) %>%
+      mutate(growthrate_1980_1990 = `1990` / `1980`) %>%
+      .[["growthrate_1980_1990"]] -> # Save as single value
+      growthrate_1980_1990
+
+    # Calculate 1980 data from 1990 using calculated growth rate
+    L144.OECD_pcflsp_Yh_wide %>%
+      mutate(`1980` = replace(`1980`, is.na(`1980`), `1990`[is.na(`1980`)] / growthrate_1980_1990)) ->
+      L144.OECD_pcflsp_Yh_wide_2
+
+    # Fill out Australia and Belgium 2004 data
+    # The ratio for USA is used as a proxy for Australia, and France for Belgium
+    L144.OECD_pcflsp_Yh_wide_2 %>%
+      filter(iso == "usa") %>%
+      mutate(value_usa_ratio = `2004` / `1998`) %>%
+      .[["value_usa_ratio"]] ->
+      value_usa_ratio
+
+    L144.OECD_pcflsp_Yh_wide_2 %>%
+      filter(iso == "fra") %>%
+      mutate(value_fra_ratio = `2004` / `2001`) %>%
+      .[["value_fra_ratio"]] ->
+      value_fra_ratio
+
+    L144.OECD_pcflsp_Yh_wide_2 %>%
+      mutate(`2004` = replace(`2004`, iso == "aus", (`1998` * value_usa_ratio)[iso == "aus"]),
+             `2004` = replace(`2004`, iso == "bel", (`2001` * value_fra_ratio)[iso == "bel"])) %>%
+      # Extrapolate the time series to all historical years
+      gather(year, value_pcflsp, -iso) %>% # Convert to long form
+      mutate(year = as.integer(year)) %>% # Convert to integer (or numeric) to extrapolate
+      group_by(iso) %>%
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(value_pcflsp = approx_fun(year, value_pcflsp, rule = 2)) %>% # Interpolation step
+      ungroup() ->
+      L144.OECD_pcflsp_Yh_final
+
+    # Other country - South Africa
+         # Other_pcflsp_m2_ctry_Yh provides residential and commercial floorspace (m2) per person for 2004 and 2005
+
+    # Time series doesn't span entire "historical" range; need to extrapolate
+    # For now, use constant floorspace outside of available time series
+    Other_pcflsp_m2_ctry_Yh %>%
+      gather(year, value_pcflsp, matches(YEAR_PATTERN)) %>% # Convert to long form
+      filter(gcam.consumer == "resid") %>%
+      mutate(year = as.integer(year)) %>% # Convert year to integer (or numeric) as needed by the interpolation function.
+      # Extrapolate to all historical years
+      select(iso, year, value_pcflsp) %>%
+      complete(year = HISTORICAL_YEARS,
+               iso = "zaf") %>%
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(value_pcflsp = approx_fun(year, value_pcflsp, rule = 2)) ->
+      L144.pcflsp_m2_otherctry_Yh_final
+
+    # Combine all available national inventories
+    L144.ALL_pcflsp_Yh <- bind_rows(L144.pcflsp_m2_chn_Yh, L144.OECD_pcflsp_Yh_final,
+                                    L144.pcflsp_m2_otherctry_Yh_final)
+
+    # Replace the USA data with 50-state-derived data
+    A44.flsp_bm2_state_res %>%
```

Comment:
I had to do some detective work, but it appears that this assumptions -level state data is the same data and file name that is written out as Level 1 data in GCAM-USA.  So it is sort of a soft-link which may or may not be kept up?  It isn't wrong, but it is confusing.  I don't know if there are other instances of this.  @bpbond Have you seen this thing in other places?  At a minimum, we should just note that in the comments here.  In the csv file, it does say source RGCAM and then an R-code file name which is not quite the current name used in GCAM-USA data system. Should this be an Issue?